packages/components/src/button/index.js

```diff
@@ -186,7 +186,7 @@ export function Button( props, ref ) {
 	return (
 		<>
 			<Tooltip
-				text={ describedBy ? describedBy : label }
+				text={ children?.length && describedBy ? describedBy : label }
```

Comment:
The `Button` component was always showing the `label` for the tooltip of icon buttons, except when a `describedBy` was defined and then it would show the description.

This doesn't seem right because it makes the label not visible.